roms_tools/setup/cdr_release.py

```diff
@@ -0,0 +1,481 @@
+            append = not times or times[-1] < end_time
+            count += int(prepend) + int(append)
+
+            self.values = [self.values] * count
+
+        return self
+
+    @abstractmethod
+    def extend_to_endpoints(self, times: list, start_time, end_time):
+        """Abstract method to extend the value series to cover given time endpoints.
+
+        Must be implemented in subclasses.
+        """
+        raise NotImplementedError()
+
+
+@dataclass
+class Flux(ValueArray):
+    """Represents a time series of non-negative flux values.
+
+    Attributes
+    ----------
+    name : str
+        Name of the flux variable.
+    values : Union[NonNegativeFloat, List[NonNegativeFloat]]
+        A constant non-negative flux or a list of non-negative flux values.
+    """
+
+    values: Union[NonNegativeFloat, List[NonNegativeFloat]]
+
+    def extend_to_endpoints(self, times: list, start_time, end_time):
+        """Extends the flux series to ensure it covers the full time interval.
+
+        - Pads with `0.0` before the first time if `start_time` is earlier.
+        - Pads with `0.0` after the last time if `end_time` is later.
+        - If no times are provided, assumes a constant flux throughout.
+
+        Parameters
+        ----------
+        times : list
+            List of datetime-like time points.
+        start_time : datetime-like
+            Start of the interval to cover.
+        end_time : datetime-like
+            End of the interval to cover.
+
+        Returns
+        -------
+        self : Flux
+            Updated instance with extended values.
+        """
+        return self._extend_scalar_series(
+            times, start_time, end_time, start_pad=0.0, end_pad=0.0
+        )
+
+
+@dataclass
+class Concentration(ValueArray):
+    """Represents a time series of tracer concentrations.
+
+    Attributes
+    ----------
+    name : str
+        Name of the tracer (e.g., 'NO3', 'DIC').
+    values : Union[float, List[float]]
+        A constant concentration or a list of concentrations over time.
+    """
+
+    def extend_to_endpoints(self, times: list, start_time, end_time):
+        """Extends the concentration series to ensure it covers the full time interval.
+
+        - Pads with the first value before the first time if `start_time` is earlier.
+        - Pads with the last value after the last time if `end_time` is later.
+        - If no times are provided, assumes a constant concentration throughout.
+
+        Parameters
+        ----------
+        times : list
+            List of datetime-like time points.
+        start_time : datetime-like
+            Start of the interval to cover.
+        end_time : datetime-like
+            End of the interval to cover.
+
+        Returns
+        -------
+        self : Concentration
+            Updated instance with extended values.
+        """
+        start_pad = self.values[0] if isinstance(self.values, list) else self.values
```

Comment:
consider a single conditional:

```py
if isinstance(self.values, list):
  start_pad = self.values[0]
  end_pad = self.values[-1]
else:
  start_pad = self.values
  end_pad = self.values
```

And, now that I see that... isn't there a bug in `end_pad` being the same as `start_pad` when it isn't a list?

Reply:
> And, now that I see that... isn't there a bug in end_pad being the same as start_pad when it isn't a list?

That's intentional. If it isn't a list, it just uses the given value as a constant in time

Reply:
Using a single conditional as of [8b2fba4](https://github.com/CWorthy-ocean/roms-tools/pull/301/commits/8b2fba44a9ad35b256def217c575c17403ead6fd)